include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -878,6 +879,47 @@ struct __brick_shift_left
     }
 };
 
+struct __brick_assign
+{
+    template <typename _T1, typename _T2>
+    void
+    operator()(const _T1& __a, _T2&& __b) const
+    {
+        ::std::get<0>(::std::forward<_T2>(__b)) = ::std::get<2>(__a);     //key
+        ::std::get<1>(::std::forward<_T2>(__b)) = ::std::get<0>(__a) + 1; //index
+    }
+};
```

Comment:
The name is too generic for the implementation; it should reflect the specifics of this very assignment.
It would also be great to have some comments explaining why get<2> is assigned to get<0> and get<0>+1 is assigned to get<1>.

Reply:
Done.